Allow packaging if `development` is in the environment

montegoulding opened the pull request:
This patch alters the condition in `_packageApplication` which ensures the development environment is available to permit the `environment` to return `development command line` as it does in `-ui` mode.

Closes #193

montegoulding committed:
Allow packaging if `development` is in the environment

This patch alters the condition in `_packageApplication` which ensures the
development environment is available to permit the `environment` to return
`development command line` as it does in `-ui` mode.

trevordevore commented:
@montegoulding Will you add a comment with an example command that packages from the command line soI can verify?

montegoulding commented:
I could add an `on startup` handler to `packager.livecodescript` if you want so it can be used to build via:

```
<path to IDE exe> -ui <path to packager.livecodescript> <path to standalone.app> [<buildprofile>]
```

trevordevore commented:
> I could add an `on startup` handler to `packager.livecodescript` if you want so it can be used to build via:
> 
> ```
> <path to IDE exe> -ui <path to packager.livecodescript> <path to standalone.app> [<buildprofile>]
> ```

That would be great. Ideally we would get to a point where standalones and installers could all be built through a service like GitHub actions.

montegoulding committed:
Implement command line packaging

This patch implements a `startup` handler in the levure packager to be
executed by an IDE executable to package a levure app.

To run the packager on the command line of an installed IDE use:

```
<path to IDE exe> \
  -ui \
  <path to levure/packager/packager.livecodescript> \
  <path to app/standalone.livecode> \
  <buildprofile>
```

montegoulding commented on packager/packager.livecodescript:
```diff
@@ -2982,3 +2993,133 @@ private function _printCharXTimes pChar, pTimes
+  switch sCommandLineStatus
+  case "complete"
+    /* completed ok but there may be warnings */
+    if revStandaloneGetWarnings() is not empty then
```
@trevordevore currently in this case the command will have actually failed at the moment because the standalone builder doesn't suppress the build warning dialog correctly when `revSetTestEnvironment` is set to true.  However, I have created an issue to resolve that. Not an issue if the standalone builder can find all your resources etc though.

trevordevore commented on framework/levure.livecodescript:
```diff
@@ -2503,7 +2511,7 @@ Returns: Name
   local tName
 
-  if the environment is "development" then
+  if "development" is not in the environment then
```
@montegoulding This check should be looking for `is in`, not `is not in`.

montegoulding replied:
Ah, goood catch @trevordevore will push a fix in a sec

trevordevore reviewed (changes requested):
One minor change.

trevordevore reviewed (approved):
LGTM. Thanks!
